### DISCUSSION TITLE
How would Github Copilot react along with VS IDE Intellicode?

### DISCUSSION THREAD

**Discussion by frozenprakash** [community/community#7510] [Copilot Conversations]
As got from response earlier Github Copilot seems to be preparing for VS IDE release too in future.
In that case, how would already available VS IDE Intellicode reach while having Github Copilot too installed?

It would be great to know the highlevel roadmap of how VS IDE Intellicode & Github Copilot will be used, together or never can install copilot on VS IDE in future?

### Update - 2022-06-23

![image](https://user-images.githubusercontent.com/16418079/175341030-89a5b738-71dd-4bbd-8c26-dc0a65b68278.png)
https://github.com/github/copilot-docs/blob/main/docs/visualstudio/gettingstarted.md#getting-started-with-github-copilot-in-visual-studio

**Comment by Urethramancer** [comment]
IntelliSense isn't AI, just code completion. Very different approach with some collisions now and then when mixed with AI.

VSCode also has IntelliSense. Try that for an idea of the state of things right now.

**Reply by Neme12** [reply]
It's IntelliCode, not IntelliSense. IntelliCode is based on AI and is similar to GitHub Copilot.

**Comment by hsenag** [comment]
Hi @frozenprakash It's a good question! We (the Copilot team) are cooperating very closely with the IntelliCode team to deliver the best possible user experience for AI driven suggestions in Visual Studio. Initially at least, it’s most likely that IntelliCode will be temporarily disabled in contexts where Copilot is active, to avoid clashes.

**Comment by fnick851** [comment]
If IntelliCode is automatically disabled, can I just delete Visual Studio IntelliCode completely if I have GitHub Copilot installed?

I wonder if Copilot will completely disable IntelliCode or only disable part of its functionality (in this case it still has value to keep IntelliCode installed)?

**Reply by markw-t** [reply]
Copilot for Visual Studio only turns off IntelliCode's [whole line completions](https://docs.microsoft.com/en-us/visualstudio/intellicode/visual-studio-whole-line-completions) when it's active.  There are several other IntelliCode features that stay active.

**Comment by EdwinKwok** [comment]
In my xxx.WebAPI app

MS IntelliCode generate this
![image](https://user-images.githubusercontent.com/2855090/160250477-1cca32dc-92a1-4094-a62e-b2ac4c33e630.png)

GH Copilot generate this 
![image](https://user-images.githubusercontent.com/2855090/160250526-bad1b52b-8166-4fae-b3e4-a082aaede66c.png)

Hope this help

**Reply by markw-t** [reply]
I think this relates to IntellICode's whole line completions rather than the IntelliSense "Quick Info" shown for this API - the quick info gives some information about the method that's selected/hovered.  IntelliCode's [whole line completions](https://docs.microsoft.com/en-us/visualstudio/intellicode/visual-studio-whole-line-completions) is a feature in Visual Studio and VSCode that predicts (grey text) up to your next line of code. As @hsenag notes, the Copilot extension for Visual Studio currently suspends that feature when it's active, to avoid collisions.   Hope that makes sense?
